libqpdf/QPDFWriter.cc

```diff
@@ -1469,16 +1467,11 @@ QPDFWriter::willFilterStream(QPDFObjectHandle stream,
                              PointerHolder<Buffer>* stream_data)
 {
     compress_stream = false;
-    is_metadata = false;
```

Comment:
This is intentional. I like to initialize references right up front so that they have predictable values even if the function throws an exception early. For example, if stream.getObjGen or stream.getDict threw an exception, moving the initialization would change the meaning.

Reply:
Makes sense